packages/core/src/tools/types.ts

```diff
@@ -55,7 +55,7 @@ export interface ToolExecutionContext<TSchemaIn extends z.ZodSchema | undefined
   mastra?: MastraUnion;
   runtimeContext: RuntimeContext;
   writer?: ToolStream<any>;
-  tracingContext?: TraceContext;
+  tracingContext: TracingContext;
```

Comment:
changing `tracingContext` on ToolExecutionContext to be a required parameter, similar to `runtimeContext`.